### PR TITLE
Add annotations for grafana dashboards

### LINKED ISSUE #11: Add annotation for relevant event into grafana
Grafana lets you add annotations that draw vertical line to notify specific events[1].

To do so, we have to log relevent event into the "event" table of influxdb[2]. 

We can do something generic by logging every step of ansible thanks to ansible hooks[3].

[1] http://docs.grafana.org/reference/annotations/
[2] http://maxchadwick.xyz/blog/grafana-influxdb-annotations
[3] https://docs.ansible.com/ansible/developing_plugins.html

### PR DESCRIPTION
**Pull request by denaitre**
fix #11

Add an Ansible callback plugin that fills an influx database (called 'events')
with entries based on Ansible events:
1. create an event for each Ansible event:
    * playbook-start
    * play-start
    * task-start
2. push the event list to influxdb at the end of playbooks

An entry looks like as the following:
```
[{'fields': {'text': 'Playbook up.yml started on frennes.rennes.grid5000.fr by
dpertin'},
'measurement': 'events',
'tags': {'hostname': 'frennes.rennes.grid5000.fr',
'type': 'playbook-start',
'user': 'dpertin'},
'time': '2017-01-19T23:24:43.069259'}]
```

To activate the plugin, the 'ansible.cfg' must be provided
(see http://docs.ansible.com/ansible/intro_configuration.html).
For instance, it can be provided as an environment variable:
```
$ ANSIBLE_CONFIG="./enos/ansible.cfg" python -m enos.enos deploy
```

Once deployed, an ssh tunnel might be used to display grafana locally:
```
$ ssh -NL 3000:<control_node>:3000 rennes.g5k
```

Then, the grafana dashboard can be displayed from http://localhost:3000/
(credentials are admin/admin by default).

A template is provided to display annotations from infludb: `enos/templates/grafana_dashboard_annotations.json`

### CODE REVIEW

File: enos/templates/grafana_dashboard_annotations.json [hunk 1]
```diff
@@ -0,0 +1,4603 @@
...
+              "format": "short",
+              "label": null,
+              "logBase": 1,
+              "max": null,
+              "min": null,
+              "show": true
+            }
+          ]
+        }
+      ]
```
Comment (rcherrueau) [inline enos/templates/grafana_dashboard_annotations.json]: Could you please add a newline at the end of the file.

File: enos/ansible/plugins/callback/influxdb_events.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+"""
+How this plugins works:
+1. Create an empty list at the beginning of playbooks
+2. Add an event to the list for each playbook/play/task
+3. Connect to influxdb at the end of playbooks and commit events
+"""
+
+class CallbackModule(CallbackBase):
+    """
+    This callback module tells you how long your plays ran for.
```
Comment (rcherrueau) [inline enos/ansible/plugins/callback/influxdb_events.py]: how long?

Comment (denaitre) [inline enos/ansible/plugins/callback/influxdb_events.py]: wrong comment

**Comment by msimonin** [comment]
We already have a dashboard that get versionned. I'm not a big fan of keeping dashboards in the tree. Could we find a way to publish dashboards elsewhere ?

**Comment by rcherrueau** [comment]
To follow @msimonin remark, I would say [BeyondTheClouds/kolla-g5k-results](https://github.com/BeyondTheClouds/kolla-g5k-results) is a good place to put the Grafana dashboard. Look at `kolla-g5k-results/files/grafana`.

**Comment by rcherrueau** [comment]
Good job, well documented, 👍

**Pull request merged**